src/claim.rs

```diff
@@ -0,0 +1,71 @@
+use serde::{Deserialize, Serialize};
+use serde_cbor::Value;
+
+/// Registered CWT claims from the
+/// [CWT Claims registry](https://www.iana.org/assignments/cwt/cwt.xhtml).
+pub trait Claim: Into<Value> {
+    fn label() -> Label;
+}
+
+pub enum Label {
+    Text(String),
+    Integer(i128),
+}
+impl From<Label> for Value {
+    fn from(label: Label) -> Value {
+        match label {
+            Label::Text(k) => Value::Text(k),
+            Label::Integer(k) => Value::Integer(k),
+        }
+    }
+}
+
+#[derive(Clone, Debug, Deserialize, Serialize, PartialEq)]
+pub enum NumericDate {
+    Integer(i128),
+    Fractional(f64),
```

Comment:
I think it could be helpful to include the units to avoid any confusion
```suggestion
    IntegerSeconds(i128),
    FractionalSeconds(f64),
```